### PR TITLE
update the usage of cryptographic nonce metadata

### LINKED ISSUE #198: Stylesheet loading should use "fetch a CSS style sheet" from CSSOM
When https://github.com/whatwg/html/issues/968 is taken care of, y'all should incorporate the changes. Alternatively, y'all could be first, setting an example for WHATWG.

### PR DESCRIPTION
**Pull request by siusin**
To fix #198 :
1. update a few sections about the cryptographic nonce metadata;
2. add the integrity attribute;

### CODE REVIEW

File: single-page.bs [hunk 1]
```diff
@@ -733,6 +734,11 @@ urlPrefix: https://tools.ietf.org/html/
     url: rfc7232#section-2.2; type: http-header; spec: rfc7232; text: last-modified
     url: rfc7234#section-5.2; type: http-header; spec: rfc7234; text: cache-control
 
+#  ************************** INFRA *************************************
```
Comment (chaals) [inline single-page.bs]: this is unnecessary, bikeshed already has this reference. (Plus it is unnecessary as noted above)

File: sections/webappapis.include [hunk 2]
```diff
@@ -2741,20 +2846,31 @@
...
           object</a>.</li>
 
-          <li><a>Create a script</a> using <var>script source</var> as the script source,
-          the [=url/URL=] where <var>script source</var> can be found, <var>scripting
-          language</var> as the scripting language, and <var>settings object</var> as the
-          <a>environment settings object</a>.</li>
+          <li>Let <var>base URL</var> be <var>initiating script</var>'s <a>base URL</a>.</li>
+
+          <li>
+          <p>Let <var>fetch options</var> be a <a>script fetch options</a> whose <a>cryptographic nonce</a> is <var>initiating script</var>'s <a>fetch options</a>'s
```
Comment (chaals) [inline sections/webappapis.include]: be a script settings options -> be a set of...

File: sections/webappapis.include [hunk 3]
```diff
@@ -59,34 +59,62 @@
...
+  A <dfn>module script</dfn> is another type of <a>script</a>. It has no additional <a>items</a>.
 
-  : A <dfn>cryptographic nonce</dfn>
-  :: A <a>cryptographic nonce</a> used to fetch imported modules.
+  An <dfn>environment</dfn> is an object that identifies the settings of a
+  current or potential execution environment. An <a>environment</a> has the following fields:
 
-  : A <dfn>parser state</dfn>
-  :: The <a>parser metadata</a> used to fetch imported modules.
+  : A <dfn>id</dfn>
```
Comment (chaals) [inline sections/webappapis.include]: A id -> An id

File: sections/webappapis.include [hunk 4]
```diff
@@ -59,34 +59,62 @@
...
+      <a>environment</a> is associated.
+      <p class="note">In the case of an <a>environment settings object</a>, this URL might be
+      distinct from the <a>environment settings object</a>'s <a>responsible
+      document</a>'s <a>URL</a>, due to mechanisms such as
+      <code>history.pushState()</code>.</p>
+
+  : A <a>target browsing context</a>
+  :: Null or a <a>service worker</a> that <a>controls</a> the <a>environment</a>.
+
+  : A <dfn>execution ready flag</dfn>
```
Comment (chaals) [inline sections/webappapis.include]: A execution -> An execution

File: sections/webappapis.include [hunk 5]
```diff
@@ -59,34 +59,62 @@
...
+  : An <dfn>error to rethrow</dfn>
+  :: A JavaScript value representing an error that will prevent evaluation from succeeding. It
+     will be re-thrown by any attempts to <a>run</a> the script.
+     <p class="note">Since this exception value is provided by the JavaScript specification, we know
+     that it is never null, so we use null to signal that no error has occurred.</p>
 
-  : A <dfn>module record</dfn>
-  :: A <a>Source Text Module Record</a> representing the parsed module, ready to be evaluated.
+  : <dfn>Fetch options</dfn>
+  :: A <a>script fetch options</a>, containing various options related to fetching this
```
Comment (chaals) [inline sections/webappapis.include]: A script fetch options -> A set of...
or 
-> A script fetch options object

File: sections/semantics-scriptings.include [hunk 6]
```diff
@@ -492,11 +492,17 @@ o............A....e
        content attribute.
   15. <p>Let <var>cryptographic nonce</var> be the element's <a href="#cryptographicnonce">[[<span>CryptographicNonce</span>]]</a>
        internal slot's value.
-  16. Let <var>parser state</var> be "<code>parser-inserted</code>" if the <{script}> element has been flagged
+  16. If the <{script}> element has an <{script/integrity}> attribute, then let <var>integrity metadata</var> be that attribute's value.
+      Otherwise, let <var>integrity metadata</var> be the empty string.
+  17. Let <var>parser metadata</var> be "<code>parser-inserted</code>" if the <{script}> element has been flagged
        as "[=parser-inserted=]", and "`not parser-inserted`" otherwise.
-  17. Let <var>settings</var> be the element's <a>node document</a>'s {{Window}} object's
+  18. Let options be a <a>script fetch options</a> whose <a>cryptographic nonce metadata</a> is
```
Comment (chaals) [inline sections/semantics-scriptings.include]: a script fetch options -> a set of script...
or -> a script fetch options object

File: sections/semantics-document-metadata.include [hunk 7]
```diff
@@ -631,26 +631,28 @@
   must run the following steps:
 
 
-  1. If the <{link/href}> attribute's value is the empty string, then abort these steps.
+  1. If the <{link/href}> attribute's value is the empty string, then return.
```
Comment (chaals) [inline sections/semantics-document-metadata.include:636]: note comment below on consistent terminology in algorithms.

**Review by chaals — CHANGES_REQUESTED** [review]
Some editorial and linking changes, but the content looks OK to me.

**Comment by siusin** [comment]
PR updated. Thanks for the review @chaals .

File: sections/changes.include [hunk 8]
```diff
@@ -24,6 +24,8 @@
             <dd>Substantive change. Fixed <a href="https://github.com/w3c/html/issues/1104">issue 1104</a></dd>
     <dt><a href="https://github.com/w3c/html/pull/1167">Added <{media/disableRemotePlayback}> to the {{HTMLMediaElement}} interface</a></dt>
             <dd>Fixed <a href="https://github.com/w3c/html/issues/1047">issue 1047</a></dd>
+    <dt><a href="https://github.com/w3c/html/pull/1177">Cleaned up the Fetch sections to apply the <a>cryptographic nonce metadata</a></a> </dt>
+            <dd>Fixed <a href="https://github.com/w3c/html/issues/198">issue 198</a></dd>
```
Comment (chaals) [inline sections/changes.include]: you also added SRI, right?

File: sections/semantics-document-metadata.include (hunk 7 above)
Comment (chaals) [reply sections/semantics-document-metadata.include:636]: No need to do anything here.

File: sections/semantics-scriptings.include [hunk 9]
```diff
@@ -431,11 +431,11 @@ o............A....e
 
       Otherwise, if the <{script}> element has a <{script/type}> attribute, let
       <var>the script block's type string</var> for this <{script}> element be the value of that
-      attribute with any leading or trailing sequences of [=space characters=] removed.
+      attribute with <a>strip leading and trailing ASCII whitespace</a>.
```
Comment (chaals) [inline sections/semantics-scriptings.include]: This line should read:

> ` attribute after <a>stripping leading and trailing white space</a>. `

to pick up the term defined in the spec.

File: sections/webappapis.include [hunk 10]
```diff
@@ -159,33 +181,116 @@
...
+   <dd><p>The <a>parser metadata</a> used for the
+   initial fetch and for fetching any imported modules</p></dd>
+
+   <dt><dfn for="script">credentials mode</dfn></dt>
+
+   <dd><p>The <a>credentials mode</a> used for the
+   initial fetch (for <a>module scripts</a>) and for fetching any
+   imported modules (for both <a>module scripts</a> and <a>classic scripts</a>)</p></dd>
+
+   <dt><dfnfor="script">referrer policy</dfn></dt>
```
Comment (chaals) [inline sections/webappapis.include]: dfn for (missing space)

File: sections/webappapis.include [hunk 11]
```diff
@@ -159,33 +181,116 @@
...
+   initial fetch and for fetching any imported modules</p></dd>
+  </dl>
+
+  <p class="note">Recall that via the <code>import()</code> feature, <a>classic scripts</a> can import <a>module scripts</a>.</p>
+
+  
+  <p>The <dfn>default classic script fetch options</dfn> are a set of<a>script fetch options</a>
+  whose <a>cryptographic nonce</a> is the empty
+  string, <a>integrity metadata</a> is the
+  empty string, <a>parser metadata</span> is "<code>not-parser-inserted</code>", <a>credentials mode</a> is "<code>omit</code>", and <a>referrer
```
Comment (chaals) [inline sections/webappapis.include]: please split the line differently so you don't have a line break inside the link `<a>referrer policy</a>`

File: sections/webappapis.include [hunk 12]
```diff
@@ -159,33 +181,116 @@
...
+
+   <dt><dfn>set up the module script request</dfn></dt>
+   <dd><p>Set <var>request</var>'s <a>cryptographic nonce
+   metadata</a> to <var>options</var>'s <a>cryptographic nonce</a>, its <a>integrity metadata</a> to <var>options</var>'s
+   <a>integrity metadata</a>, its <a>parser metadata</a> to <var>options</var>'s 
+   <a>parser metadata</a>, its <a>credentials mode</a> to <var>options</var>'s <a>credentials mode</a>, 
+   and its <a>referrer policy</a> to <var>options</var>'s <a>referrer policy</a>.</p></dd>
+  </dl>
+
+  <p>For any given <a>script fetch options</a> <var>options</var>, the <dfn>descendant script
```
Comment (chaals) [inline sections/webappapis.include]: please change line breaks so the `dfn` isn't split over two lines

File: sections/webappapis.include [hunk 13]
```diff
@@ -159,33 +181,116 @@
...
+  fetch options</dfn> are a new set of <a>script fetch options</a> whose <a>items</a> all have the same values,
+   except for the <a>integrity metadata</a>, which is instead the
+  empty string.</p>
+
+  <hr>
+
+  <p>The algorithms below can be customized by optionally supplying a custom <dfn for="fetching scripts">perform the
+  fetch</dfn> hook, which takes a <a>request</a> and an <dfn for="fetching scripts"><var>is top-level</var></dfn> flag. The algorithm must complete with a <a>response</a> (which may be a <a>network error</a>), either
```
Comment (chaals) [inline sections/webappapis.include]: line breaks...

File: sections/webappapis.include [hunk 14]
```diff
@@ -217,20 +322,22 @@
        flag to the <a>create a classic script</a> algorithm as well.
   11. Asynchronously complete this algorithm with <var>script</var>.
 
-  To <dfn>fetch a classic worker script</dfn> given a <var>url</var>, a <var>referrer</var>, a
-  <var>settings object</var>, and a <var>destination</var>, run these steps. The algorithm will
+  To <dfn>fetch a classic worker script</dfn> given a <var>url</var>, a <var>fetch
```
Comment (chaals) [inline sections/webappapis.include]: line breaks...

File: sections/webappapis.include [hunk 15]
```diff
@@ -217,20 +322,22 @@
...
-      "<code>script</code>", <a>destination</a> is <var>destination</var>, <a>referrer</a> is
-      <var>referrer</var>, <a for="request">mode</a> is "<code>same-origin</code>",
-      <a>credentials mode</a> is "<code>same-origin</code>", <a>parser metadata</a> is
-      "<code>not parser-inserted</code>", and whose <a>use-URL-credentials flag</a> is set.
-  2. If the caller specified custom steps to <a>set up the request</a>, perform them on
-      <var>request</var>.
-  3. <a>Fetch</a> <var>request</var>.
+  1. Let <var>request</var> be a new <a>request</a> whose <a>url</a> is <var>url</var>, 
+      <a>client</a> is <var>fetch client settings object</var>, <a>destination</a> is <var>destination</var>,
+       <a>mode</a> is "<code>same-origin</code>", <a>credentials mode</a> is "<code>same-origin</code>", <a>parser
```
Comment (chaals) [inline sections/webappapis.include]: line breaks

File: sections/webappapis.include [hunk 16]
```diff
@@ -217,20 +322,22 @@
...
+       metadata</a> is "<code>not parser-inserted</code>", and whose
+       <a>use-URL-credentials flag</a> is set.
+  2. If the caller specified custom steps to <a>perform
+      the fetch</a>, perform them on <var>request</var>, with the <a>is top-level</a> flag set. Return from this algorithm,
+      and when the custom <a>perform the fetch</span> steps
+      complete with <a>response</a> <var>response</var>, run the
+      remaining steps.
+      <p>Otherwise, <a>fetch</a> <var>request</var>.</p>
   4. Return from this algorithm, and run the remaining steps as part of the fetch's <a>process
```
Comment (chaals) [inline sections/webappapis.include]: while you are here, can you lease fix this line break too?

File: sections/webappapis.include [hunk 17]
```diff
@@ -313,9 +419,9 @@
       <a>module script</a>, asynchronously complete this algorithm with <var>module script</var>.
 
   To <dfn lt="fetching a single module script|fetch a single module script">fetch a single module script</dfn>,
-  given a <var>url</var>, a
-  <var>credentials mode</var>, a <var>cryptographic nonce</var>, a <var>parser state</var>,
-  a <var>destination</var>, and a <var>settings object</var>, run these steps. The algorithm will
+  given a <var>url</var>, a <var>fetch client
```
Comment (chaals) [inline sections/webappapis.include]: line breaks for 2 lines in a row

File: sections/webappapis.include [hunk 18]
```diff
@@ -326,12 +432,11 @@
...
       "<code>fetching</code>".
-  5. Let <var>request</var> be a new <a>request</a> whose <a for="request">url</a> is
-      <var>url</var>, <a>destination</a> is <var>destination</var>, <a for="request">type</a> is
-      "<code>script</code>", <a for="request">mode</a> is "<code>cors</code>",
-      <a>credentials mode</a> is <var>credentials mode</var>, <a>cryptographic nonce metadata</a>
-      is <var>cryptographic nonce</var>, <a>parser metadata</a> is <var>parser state</var> and
-      <a>client</a> is <var>settings object</var>.
+  5. Let <var>request</var> be a new <a>request</a> whose
+      <a>url</a> is <var>url</var>, <a>destination</a> is <var>destination</var>, <a>mode</a> is "<code>cors</code>", 
+      <a>referrer</a> is <var>referrer</var>, and <a>client</a> is <var>fetch client settings
```
Comment (chaals) [inline sections/webappapis.include:434]: line breaks

File: sections/webappapis.include [hunk 19]
```diff
@@ -159,33 +181,116 @@
...
+
+  <p>The algorithms below can be customized by optionally supplying a custom <dfn for="fetching scripts">perform the
+  fetch</dfn> hook, which takes a <a>request</a> and an <dfn for="fetching scripts"><var>is top-level</var></dfn> flag. The algorithm must complete with a <a>response</a> (which may be a <a>network error</a>), either
+  synchronously (when using <a>fetch a classic worker-imported script</a>) or asynchronously
+  (otherwise). The <var>is top-level</var> flag will be set
+  for all <a>classic script</a> fetches, and for the initial fetch when <a>fetching a module script tree</a> or <a>fetching a module worker script tree</a>, but not for the fetches resulting from
+  <code>import</code> statements encountered throughout the graph.</p>
+
+  <div class="note">
+   <p>By default, not supplying the <a>perform the
```
Comment (chaals) [inline sections/webappapis.include]: line breaks

File: sections/webappapis.include [hunk 20]
```diff
@@ -159,33 +181,116 @@
...
+  empty string, <a>parser metadata</span> is "<code>not-parser-inserted</code>", <a>credentials mode</a> is "<code>omit</code>", and <a>referrer
+  policy</a> is the empty string.</p>
+
+  <p>Given a <a>request</a> <var>request</var> and a <a>script
+  fetch options</a> <var>options</var>, we define:</p>
+
+  <dl>
+   <dt><dfn>set up the classic script request</dfn></dt>
+   <dd><p>Set <var>request</var>'s <a>cryptographic nonce
+   metadata</a> to <var>options</var>'s <a>cryptographic nonce</a>, its <a>integrity metadata</a> to <var>options</var>'s
```
Comment (chaals) [inline sections/webappapis.include]: remove the final "s" at the end of the line - should read

> `...<var>options</var>' `

File: sections/webappapis.include [hunk 21]
```diff
@@ -159,33 +181,116 @@
...
+  <p class="note">Recall that via the <code>import()</code> feature, <a>classic scripts</a> can import <a>module scripts</a>.</p>
+
+  
+  <p>The <dfn>default classic script fetch options</dfn> are a set of<a>script fetch options</a>
+  whose <a>cryptographic nonce</a> is the empty
+  string, <a>integrity metadata</a> is the
+  empty string, <a>parser metadata</span> is "<code>not-parser-inserted</code>", <a>credentials mode</a> is "<code>omit</code>", and <a>referrer
+  policy</a> is the empty string.</p>
+
+  <p>Given a <a>request</a> <var>request</var> and a <a>script
```
Comment (chaals) [inline sections/webappapis.include]: line breaks

File: sections/webappapis.include [hunk 22]
```diff
@@ -59,34 +59,62 @@
...
-  : A <dfn>parser state</dfn>
-  :: The <a>parser metadata</a> used to fetch imported modules.
+  : An <dfn>id</dfn>
+  :: An opaque string that uniquely identifies the <a>environment</a>.
+
+  : A <dfn>creation URL</dfn>
+  :: A <a>URL record</a> that represents the location of the resource with which the
+      <a>environment</a> is associated.
+      <p class="note">In the case of an <a>environment settings object</a>, this URL might be
+      distinct from the <a>environment settings object</a>'s <a>responsible
```
Comment (chaals) [inline sections/webappapis.include]: please don't split the link over 2 different lines

File: sections/webappapis.include [hunk 23]
```diff
@@ -59,34 +59,62 @@
...
-
   : A <dfn>Source text</dfn>
   :: A string containing a block of executable code to be evaluated as a JavaScript
       <a for="ecma">Script</a>.
 
-  : Optionally, a <dfn>muted errors</dfn> flag
-  :: A flag which, if set, means that error information will not be provided for errors in this
-    script (used to mute errors for cross-origin scripts, since that can leak private information).
+  : A <dfn>record</dfn>
+  :: Either a <a>Script Record</a>, for <a>classic
```
Comment (chaals) [inline sections/webappapis.include]: please don't split the link over 2 different lines

File: sections/semantics-document-metadata.include [hunk 24]
```diff
@@ -449,7 +449,7 @@
   The <dfn element-attr for="link"><code>media</code></dfn> attribute says which media the resource
   applies to. The value must be a <a>valid media query list</a>.
 
-  <!-- def <{link/integrity}> -->
+  The <dfn element-attr for="link"><code>integrity</code></dfn> attribute represents the <a>integrity metadata</a> for requests which this element is responsible for. The value is text. The attribute must not be specified on <{link}> elements that do not have a <{link/rel}> attribute that contains the <{link/stylesheet}> keyword. [[!SRI]]
```
Comment (chaals) [inline sections/semantics-document-metadata.include]: please split this line for easier reading in tools that don't auto-wrap. E.g. < 100 characters / line ...

**Review by chaals — CHANGES_REQUESTED** [review]
Thanks for the update.

Unfortunately there are a handful more editorial and nitpicky things - it seems I forgot to add many of my comments to the review before, and found some I had missed

Also, at https://w3c.github.io/html/dom.html#global-attributes the nonce attribute doesn't link to a definition, but it should...

**Review by chaals — APPROVED** [review]

**Pull request merged**